rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -18,19 +21,48 @@
 from rotkehlchen.types import ChecksumEvmAddress
 from rotkehlchen.user_messages import MessagesAggregator
 
+from .constants import CPT_LIQUITY
 
 if TYPE_CHECKING:
     from rotkehlchen.assets.asset import Asset
     from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
     from rotkehlchen.chain.evm.contracts import EvmContract
     from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.db.drivers.gevent import DBCursor
 
 MIN_COLL_RATE = '1.1'
 
 logger = logging.getLogger(__name__)
 log = RotkehlchenLogsAdapter(logger)
 
 
+QUERY_STAKING_EVENTS = """
+WHERE event_identifier IN
+(SELECT A.event_identifier FROM history_events AS A JOIN history_events AS B ON A.event_identifier = B.event_identifier WHERE
```

Comment:
Won't it be possible to use a db filter here?

Reply:
I believe the filter will over complicated it since is a query that we won't reuse and it has many special conditions